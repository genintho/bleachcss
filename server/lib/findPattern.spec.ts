import { findPattern } from "./findPattern";
import testData from "./__snapshots__/patternUrls";

test("findPattern should work", () => {
	expect(
		testData.map((url) => {
			return findPattern(url);
		})
	).toMatchSnapshot();
});
