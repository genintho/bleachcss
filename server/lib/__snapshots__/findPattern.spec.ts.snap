// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`findPattern should work 1`] = `
Array [
  Object {
    "name": "sumome-140a.kxcdn.<domain>/virtual/<hash40>/client/css/sme-popup.css",
    "pattern": "sumome-140a\\\\.kxcdn(.(ac|co|com|edu|gov|me|net|org))?.[^./]+/virtual/[0-9a-f]{40}/client/css/sme-popup\\\\.css",
    "url": "http://sumome-140a.kxcdn.com/virtual/3056843ea62d1018df1eb76be85ed1b4ad13fc11/client/css/sme-popup.css",
  },
  Object {
    "name": "ucarecdn.<domain>/assets/frontend/public/fonts.<hash32>.css",
    "pattern": "ucarecdn(.(ac|co|com|edu|gov|me|net|org))?.[^./]+/assets/frontend/public/fonts\\\\.[0-9a-f]{32}\\\\.css",
    "url": "https://ucarecdn.com/assets/frontend/public/fonts.75d2bbdf1740f43006f62ed6d103584d.css",
  },
  Object {
    "name": "s.huffpost.<domain>/assets/css.php?f=_quickread.css&v=<version>",
    "pattern": "s\\\\.huffpost(.(ac|co|com|edu|gov|me|net|org))?.[^./]+/assets/css\\\\.php\\\\?f=_quickread\\\\.css&v=([\\\\d.]|[a-f\\\\d])+",
    "url": "http://s.huffpost.com/assets/css.php?f=_quickread.css&v=1475694796",
  },
  Object {
    "name": "cdn.wedding-spot.<domain>/css/bootstrap3/bootstrap.min.<hash12>.css",
    "pattern": "cdn\\\\.wedding-spot(.(ac|co|com|edu|gov|me|net|org))?.[^./]+/css/bootstrap3/bootstrap\\\\.min\\\\.[0-9a-f]{12}\\\\.css",
    "url": "http://cdn.wedding-spot.com/css/bootstrap3/bootstrap.min.3ab3438f85ad.css",
  },
  Object {
    "name": "cdn.wedding-spot.<domain>/css/bootstrap3/bootstrap.min.<hash>.css",
    "pattern": "cdn\\\\.wedding-spot(.(ac|co|com|edu|gov|me|net|org))?.[^./]+/css/bootstrap3/bootstrap\\\\.min\\\\.[0-9a-f]{10,}\\\\.css",
    "url": "http://cdn.wedding-spot.com/css/bootstrap3/bootstrap.min.3056843ea62d1018df1eb76be85ed1b4a.css",
  },
  Object {
    "name": "cdn-asset-mel-1.airsquare.<domain>/www/site.css?timestamp=<timestamp>",
    "pattern": "cdn-asset-mel-1\\\\.airsquare(.(ac|co|com|edu|gov|me|net|org))?.[^./]+/www/site\\\\.css\\\\?timestamp=[\\\\d]+",
    "url": "https://cdn-asset-mel-1.airsquare.com/www/site.css?timestamp=20160223234310",
  },
  Object {
    "name": "assets.gothamistllc.<domain>/v8/css/sfist.css",
    "pattern": "assets\\\\.gothamistllc(.(ac|co|com|edu|gov|me|net|org))?.[^./]+/v8/css/sfist\\\\.css",
    "url": "http://assets.gothamistllc.com/v8/css/sfist.css",
  },
  Object {
    "name": "blog.rudylee.<domain>/stylesheets/screen.css",
    "pattern": "blog\\\\.rudylee(.(ac|co|com|edu|gov|me|net|org))?.[^./]+/stylesheets/screen\\\\.css",
    "url": "http://blog.rudylee.com/stylesheets/screen.css",
  },
  Object {
    "name": "fathomaway.<domain>/static/cache/css/<hash12>.css",
    "pattern": "fathomaway(.(ac|co|com|edu|gov|me|net|org))?.[^./]+/static/cache/css/[0-9a-f]{12}\\\\.css",
    "url": "http://fathomaway.com/static/cache/css/0b3ff9753548.css",
  },
  Object {
    "name": "s.huffpost.<domain>/assets/css.php?f=_quickread.css&v=<version>",
    "pattern": "s\\\\.huffpost(.(ac|co|com|edu|gov|me|net|org))?.[^./]+/assets/css\\\\.php\\\\?f=_quickread\\\\.css&v=([\\\\d.]|[a-f\\\\d])+",
    "url": "http://s.huffpost.com/assets/css.php?f=_quickread.css&v=1475694796",
  },
  Object {
    "name": "s.huffpost.<domain>/assets/css.php?f=hptwitter_anywhere.css&9",
    "pattern": "s\\\\.huffpost(.(ac|co|com|edu|gov|me|net|org))?.[^./]+/assets/css\\\\.php\\\\?f=hptwitter_anywhere\\\\.css&9",
    "url": "http://s.huffpost.com/assets/css.php?f=hptwitter_anywhere.css&9",
  },
  Object {
    "name": "s.huffpost.<domain>/assets/css.php?v=<version>&f=hp_modules%2Fmodule.bubble.css%2Chp_modules%2Fmodule.sponsor-widgets.css%2Chp_modules%2Fgeneral.css%2Chp_modules%2Fmodule.fb_like_button.css%2Cfeatured-posts.css%2Clogos.css%2Ccomments.css%2Cmod-follow.css%2Cfloating_share.css",
    "pattern": "s\\\\.huffpost(.(ac|co|com|edu|gov|me|net|org))?.[^./]+/assets/css\\\\.php\\\\?v=([\\\\d.]|[a-f\\\\d])+&f=hp_modules%2Fmodule\\\\.bubble\\\\.css%2Chp_modules%2Fmodule\\\\.sponsor-widgets\\\\.css%2Chp_modules%2Fgeneral\\\\.css%2Chp_modules%2Fmodule\\\\.fb_like_button\\\\.css%2Cfeatured-posts\\\\.css%2Clogos\\\\.css%2Ccomments\\\\.css%2Cmod-follow\\\\.css%2Cfloating_share\\\\.css",
    "url": "http://s.huffpost.com/assets/css.php?v=1475694796&f=hp_modules%2Fmodule.bubble.css%2Chp_modules%2Fmodule.sponsor-widgets.css%2Chp_modules%2Fgeneral.css%2Chp_modules%2Fmodule.fb_like_button.css%2Cfeatured-posts.css%2Clogos.css%2Ccomments.css%2Cmod-follow.css%2Cfloating_share.css",
  },
  Object {
    "name": "s.huffpost.<domain>/assets/css.php?v=<version>&f=verticals%2Fdefault.css%2Ctopnav_new%2Fmodule.edition_nav.css%2Cbasic.css%2Cmasthead.css%2Call_pages.css%2Cmaster.css%2Cbuttons.css%2Chp_modules%2Fmodule.photo_galleries.css%2Cprofiles.css%2Cadd%2Finner.css%2Chp_modules%2Fmodule.threeup.css%2Cemail_share.css%2Ccommercial.css%2Csite_promo.css%2Cpoll_like.css",
    "pattern": "s\\\\.huffpost(.(ac|co|com|edu|gov|me|net|org))?.[^./]+/assets/css\\\\.php\\\\?v=([\\\\d.]|[a-f\\\\d])+&f=verticals%2Fdefault\\\\.css%2Ctopnav_new%2Fmodule\\\\.edition_nav\\\\.css%2Cbasic\\\\.css%2Cmasthead\\\\.css%2Call_pages\\\\.css%2Cmaster\\\\.css%2Cbuttons\\\\.css%2Chp_modules%2Fmodule\\\\.photo_galleries\\\\.css%2Cprofiles\\\\.css%2Cadd%2Finner\\\\.css%2Chp_modules%2Fmodule\\\\.threeup\\\\.css%2Cemail_share\\\\.css%2Ccommercial\\\\.css%2Csite_promo\\\\.css%2Cpoll_like\\\\.css",
    "url": "http://s.huffpost.com/assets/css.php?v=1475694796&f=verticals%2Fdefault.css%2Ctopnav_new%2Fmodule.edition_nav.css%2Cbasic.css%2Cmasthead.css%2Call_pages.css%2Cmaster.css%2Cbuttons.css%2Chp_modules%2Fmodule.photo_galleries.css%2Cprofiles.css%2Cadd%2Finner.css%2Chp_modules%2Fmodule.threeup.css%2Cemail_share.css%2Ccommercial.css%2Csite_promo.css%2Cpoll_like.css",
  },
  Object {
    "name": "static1.squarespace.<domain>/static/sitecss/<hash>/222/<hash>/<hash>/2755-05142015/<hash>/site.css?&filterFeatures=false",
    "pattern": "static1\\\\.squarespace(.(ac|co|com|edu|gov|me|net|org))?.[^./]+/static/sitecss/[0-9a-f]{10,}/222/[0-9a-f]{10,}/[0-9a-f]{10,}/2755-05142015/[0-9a-f]{10,}/site\\\\.css\\\\?&filterFeatures=false",
    "url": "http://static1.squarespace.com/static/sitecss/54f4410ce4b0a87b1ae08e97/222/515c7bd0e4b054dae3fcf003/54f4410ce4b0a87b1ae08ea0/2755-05142015/1446934641122/site.css?&filterFeatures=false",
  },
  Object {
    "name": "sumome-140a.kxcdn.<domain>/virtual/<hash40>/client/css/sme-popup.css",
    "pattern": "sumome-140a\\\\.kxcdn(.(ac|co|com|edu|gov|me|net|org))?.[^./]+/virtual/[0-9a-f]{40}/client/css/sme-popup\\\\.css",
    "url": "http://sumome-140a.kxcdn.com/virtual/3056843ea62d1018df1eb76be85ed1b4ad13fc11/client/css/sme-popup.css",
  },
  Object {
    "name": "sumome-140a.kxcdn.<domain>/virtual/<hash40>/client/css/listbuilder-popup.css",
    "pattern": "sumome-140a\\\\.kxcdn(.(ac|co|com|edu|gov|me|net|org))?.[^./]+/virtual/[0-9a-f]{40}/client/css/listbuilder-popup\\\\.css",
    "url": "http://sumome-140a.kxcdn.com/virtual/515f139915124485b5bece46bdf866a9d282df35/client/css/listbuilder-popup.css",
  },
  Object {
    "name": "sumome-140a.kxcdn.<domain>/virtual/<hash40>/client/css/sumome-welcome-popup.css",
    "pattern": "sumome-140a\\\\.kxcdn(.(ac|co|com|edu|gov|me|net|org))?.[^./]+/virtual/[0-9a-f]{40}/client/css/sumome-welcome-popup\\\\.css",
    "url": "http://sumome-140a.kxcdn.com/virtual/84ad0fa1b1bdc9a5580e76676f84f42bfbd37f43/client/css/sumome-welcome-popup.css",
  },
  Object {
    "name": "sumome-140a.kxcdn.<domain>/virtual/<hash40>/client/css/sumome-share-client.css",
    "pattern": "sumome-140a\\\\.kxcdn(.(ac|co|com|edu|gov|me|net|org))?.[^./]+/virtual/[0-9a-f]{40}/client/css/sumome-share-client\\\\.css",
    "url": "http://sumome-140a.kxcdn.com/virtual/95cc4e503a7a593cfc07f72b32f15ea2118f82cf/client/css/sumome-share-client.css",
  },
  Object {
    "name": "sumome-140a.kxcdn.<domain>/virtual/<hash40>/client/css/sumome-scrollbox-popup.css",
    "pattern": "sumome-140a\\\\.kxcdn(.(ac|co|com|edu|gov|me|net|org))?.[^./]+/virtual/[0-9a-f]{40}/client/css/sumome-scrollbox-popup\\\\.css",
    "url": "http://sumome-140a.kxcdn.com/virtual/a51255e7f5786ecb71277f59babb0e1367f03305/client/css/sumome-scrollbox-popup.css",
  },
  Object {
    "name": "searchwp-live-ajax-search Wordpress Plugin",
    "pattern": "/wp-content/plugins/(.+?)/",
    "url": "http://www.nomadicmatt.com/wp-content/plugins/searchwp-live-ajax-search/assets/styles/style.css?ver=1.2.0",
  },
  Object {
    "name": "woocommerce Wordpress Plugin",
    "pattern": "/wp-content/plugins/(.+?)/",
    "url": "http://www.nomadicmatt.com/wp-content/plugins/woocommerce/assets/css/woocommerce-layout.css?ver=2.6.14",
  },
  Object {
    "name": "nomadicmatt Wordpress Plugin",
    "pattern": "/wp-content/themes/(.+?)/",
    "url": "http://www.nomadicmatt.com/wp-content/themes/nomadicmatt/style.css?ver=1.0.17",
  },
  Object {
    "name": "assets-cdn.github.<domain>/assets/github-<hash64>.css",
    "pattern": "assets-cdn\\\\.github(.(ac|co|com|edu|gov|me|net|org))?.[^./]+/assets/github-[0-9a-f]{64}\\\\.css",
    "url": "https://assets-cdn.github.com/assets/github-a20b0761ae9a52b4ed0d75922904eb8ffb5934a855914ec141ff3b2140a18b9a.css",
  },
  Object {
    "name": "assets.pixolum.<domain>/css/main.min_14a.css",
    "pattern": "assets\\\\.pixolum(.(ac|co|com|edu|gov|me|net|org))?.[^./]+/css/main\\\\.min_14a\\\\.css",
    "url": "https://assets.pixolum.com/css/main.min_14a.css",
  },
  Object {
    "name": "autodocs.<domain>/assets/application-<hash64>.css",
    "pattern": "autodocs(.(ac|co|com|edu|gov|me|net|org))?.[^./]+/assets/application-[0-9a-f]{64}\\\\.css",
    "url": "https://autodocs.io/assets/application-6d74911a7dbcf02c27ac5d8223626a2870913c23a5d6f3387d2ff42440b0dfcd.css",
  },
  Object {
    "name": "cdn-asset-mel-1.airsquare.<domain>/www/site.css?timestamp=<timestamp>",
    "pattern": "cdn-asset-mel-1\\\\.airsquare(.(ac|co|com|edu|gov|me|net|org))?.[^./]+/www/site\\\\.css\\\\?timestamp=[\\\\d]+",
    "url": "https://cdn-asset-mel-1.airsquare.com/www/site.css?timestamp=20160223234310",
  },
  Object {
    "name": "cdn.sstatic.<domain>/Sites/stackoverflow/all.css?v=<version>",
    "pattern": "cdn\\\\.sstatic(.(ac|co|com|edu|gov|me|net|org))?.[^./]+/Sites/stackoverflow/all\\\\.css\\\\?v=([\\\\d.]|[a-f\\\\d])+",
    "url": "https://cdn.sstatic.net/Sites/stackoverflow/all.css?v=df272f6aeb12",
  },
  Object {
    "name": "ucarecdn.<domain>/assets/frontend/public/app.<hash32>.css",
    "pattern": "ucarecdn(.(ac|co|com|edu|gov|me|net|org))?.[^./]+/assets/frontend/public/app\\\\.[0-9a-f]{32}\\\\.css",
    "url": "https://ucarecdn.com/assets/frontend/public/app.afafa286b5dfc328f84ac23e4c937cb0.css",
  },
  Object {
    "name": "wfaprod-1290.kxcdn.<domain>/themes.css?v=<version>",
    "pattern": "wfaprod-1290\\\\.kxcdn(.(ac|co|com|edu|gov|me|net|org))?.[^./]+/themes\\\\.css\\\\?v=([\\\\d.]|[a-f\\\\d])+",
    "url": "https://wfaprod-1290.kxcdn.com/themes.css?v=1",
  },
  Object {
    "name": "www.google.<domain>/uds/api/visualization/1.1/<hash32>/ui+entable+en.css",
    "pattern": "www\\\\.google(.(ac|co|com|edu|gov|me|net|org))?.[^./]+/uds/api/visualization/1\\\\.1/[0-9a-f]{32}/ui\\\\+entable\\\\+en\\\\.css",
    "url": "https://www.google.com/uds/api/visualization/1.1/676331e4f5be6ab26a5ba55c06d0ef71/ui+entable+en.css",
  },
  Object {
    "name": "caphunters.<domain>/themes/caphunters/cache/v_92_<hash32>_all.css",
    "pattern": "caphunters(.(ac|co|com|edu|gov|me|net|org))?.[^./]+/themes/caphunters/cache/v_92_[0-9a-f]{32}_all\\\\.css",
    "url": "https://caphunters.es/themes/caphunters/cache/v_92_66118a9fc525536c607dd17aa89388ff_all.css",
  },
  Object {
    "name": "testdomain.<domain>/test.css",
    "pattern": "testdomain(.(ac|co|com|edu|gov|me|net|org))?.[^./]+/test\\\\.css",
    "url": "https://testdomain.co.uk/test.css",
  },
  Object {
    "name": "testdomain.<domain>/test.css",
    "pattern": "testdomain(.(ac|co|com|edu|gov|me|net|org))?.[^./]+/test\\\\.css",
    "url": "https://testdomain.uk/test.css",
  },
]
`;
